File: kanito/Classi/CustomControls/MyButton/MyButton.m
//
//  MyButton.m
//  csen cinofilia
//
//  Created by Luciano Calderano on 05/07/16.
//  Copyright © 2016 Kanito. All rights reserved.
//

#import "MyButton.h"

@implementation MyButton

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self defaultMyButton];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self defaultMyButton];
    }
    return self;
}

- (void) defaultMyButton {
    NSString *s = [self titleForState:UIControlStateNormal];
    if ([[s substringToIndex:1] isEqualToString:@"#"])
        [self setTitle:keyLang([s substringFromIndex:1]) forState:UIControlStateNormal];
    self.layer.masksToBounds = NO;
    self.showsTouchWhenHighlighted = YES;
    self.titleLabel.font = [MyFont fontSize:self.titleLabel.font.pointSize];
    if (self.backgroundColor == nil) {
        self.backgroundColor = [MyColor myOrange];
        [self setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    }
}

@end

File: Swift/Cls/CustomClass/MYButton.swift
//
//  MYButton.swift
//  Enci
//
//  Created by Luciano Calderano on 03/11/16.
//  Copyright © 2016 Kanito. All rights reserved.
//

import UIKit

class MYButton: UIButton {
    @IBInspectable let cornerRadius:CGFloat = 5.0

    @IBInspectable var borderMargins:Bool {
        get { return self.layer.borderWidth > 0 }
        set { self.layer.borderWidth = newValue ? 1 : 0; self.layer.borderColor = UIColor.lightGray.cgColor }
    }
    var title: String {
        get { return self.titleLabel!.text! }
        set { self.setTitle(newValue.tryLang(), for: UIControlState()) }
    }
    
    required internal init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.initialize()
    }
    
    private func initialize () {
        self.layer.masksToBounds = false
        
        self.showsTouchWhenHighlighted = true
        if self.backgroundColor == nil {
            self.backgroundColor = UIColor.myOrange
        }
        self.setTitleColor(UIColor.white, for: UIControlState.normal)
        self.titleLabel?.font = UIFont.mySize((self.titleLabel?.font.pointSize)!)
        self.title = self.currentTitle!
        
        self.borderMargins = false
        self.layer.cornerRadius = cornerRadius
    }
}

